### PR TITLE
Start adding irmin with eio

### PR DESCRIPTION
**Pull request by ulrikstrid**

### CODE REVIEW

**Comment by anmonteiro** [comment]
Should go in ocaml5.nix

File: ocaml/ocaml5.nix [hunk 1]
```diff
@@ -36,6 +36,93 @@ with oself;
...
+
+  irmin-eio = buildDunePackage {
+    pname = "irmin";
+
+    version = "2023-08-25+eio";
+
+    src = fetchFromGitHub {
+      owner = "mirage";
+      repo = "irmin";
+      rev = "5eb94d774e505ee9f870bbfdbc799b07439a489d";
```
Comment (anmonteiro) [inline ocaml/ocaml5.nix:49]: shouldn't this be overriding the entire irmin package? and use `inherit (irmin) src;` here?

Comment (anmonteiro) [reply ocaml/ocaml5.nix:49]: I thought `irmin-eio` was another package. I get it now.

Can you add a comment saying which branch this commit comes from?

**Comment by anmonteiro** [comment]
ah, this will need to be updated to eio v0.12

**Comment by ulrikstrid** [comment]
Since I'm not using this I'll just close in hopes that someone actually fixes irmin + eio and we can add it then

**Pull request closed without merging**